### ISSUE TITLE
BUTTONS - Add self destruct action

### ISSUE THREAD

**Issue by dxcore35** [mProjectsCode/obsidian-meta-bind-plugin#303]
### Is your Feature Request Related to a Problem?

In this list ![Screenshot 2024-04-22 at 18 10 19](https://github.com/mProjectsCode/obsidian-meta-bind-plugin/assets/7279156/430befcf-6fb6-4137-aa2b-bd9769c66da2)

Add auto destruct button action

### Describe the Feature you'd Like

After clicking on the button the button will be removed from note. It should respect the order. For example I will asign button to export to PDF and then delete itself. The deletion should be executed the last.

**Comment by mProjectsCode** [comment]
You can already achieve that with `replaceSelf` and an empty replacement string.

**Comment by dxcore35** [comment]
I just rested it unfortunately it is not working in my case for INLINE BUTTON.

```
ButtonMDRC >> error while running action Error: Replace self action not supported for inline buttons
    at bp.runReplaceSelfAction (plugin:obsidian-meta-bind-plugin:53:52)
    at bp.runAction (plugin:obsidian-meta-bind-plugin:51:7266)
    at bp.runButtonAction (plugin:obsidian-meta-bind-plugin:51:5233)
    at async onClick (plugin:obsidian-meta-bind-plugin:57:14387)
    at async f (plugin:obsidian-meta-bind-plugin:57:13107) 
```

**Comment by mProjectsCode** [comment]
Yep, because of a limitation in the Obsidian API. There is no way for inline buttons to know where in the file they are, and they need that to replace themselves.

**Comment by dxcore35** [comment]
But inline buttons are just text no? If you know the name of button that was triggered you can simply replace the inline button as text with null no?

**Comment by mProjectsCode** [comment]
The same string of text might occur multiple times in the file, or even as part of something else which is not an inline button declaration. So doing a simple find and replace could result in undesirable behavior.

**Comment by dxcore35** [comment]
Do you really thing somebody will add text in those brackets? `BUTTON[action-summarize]`? I highly doubt and other inline will also not use BUTTON as prefix. It is something to think about, I think percentage is extremely small.... Maybe just write good explanation with warning in description? will be sufficient. 

This inline buttons are gold it is pitty to not have option to destroy them :)

**Comment by Tejeev** [comment]
The Buttons plugin generates unique identifiers for each button.

**Comment by Shurikus-crypto** [comment]
@mProjectsCode wrote that the buttons can be remove using JS. Please show me an example of how to do this?
